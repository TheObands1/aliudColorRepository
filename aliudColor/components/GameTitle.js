import React from "react";
import { Text, View, StyleSheet } from "react-native";

const GameTitle = () => {
    return(
        <View style={{ flexDirection: "row" }}>
            <Text style={[TitleStyle.fontStyle, { color: "#ecf0f1" }]}>aliud</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#E64C3C" }, { textTransform: 'uppercase'}]}>c</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#E57E31" }]}>o</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#F1C431" }]}>l</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#68CC73" }]}>o</Text>
            <Text style={[TitleStyle.fontStyle, { color: "#3998DB" }]}>r</Text>
        </View>
    );
}

const TitleStyle = StyleSheet.create({
    fontStyle: {
      fontSize: 50,
      fontFamily: "dogbyte"
    }
   });

export { GameTitle };