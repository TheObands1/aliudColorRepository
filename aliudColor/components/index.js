export * from './GameTitle'
export * from "./GameplayComponents/GameGrid"