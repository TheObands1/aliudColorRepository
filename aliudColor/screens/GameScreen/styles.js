import { StyleSheet, Dimensions } from "react-native";

export default StyleSheet.create({
 backgroundStyle: {
   flex: 1,
   backgroundColor: "#0a0a0a",
   justifyContent: "center",
   alignItems: "center"
 },
 upperContainer: {
  height: Dimensions.get("window").height * 0.15,
},
 gridContainer: {
  flex: 5, 
  width:  Dimensions.get('window').width * 0.875, 
  flexDirection: 'row'
 },
 gridColumn:{
  flex: 1, 
  flexDirection: 'column' 
 },
 gridRow: {
  flex: 1
 },
 bottomContainer: {
  flex: 1,
  //marginTop: 90,
  //marginBottom: 40,
  flexDirection: 'row',
  justifyContent: "space-evenly",
  alignItems: "center"
},
bottomSection: {
  flex: 1,
},
counterCount: {
  fontFamily: 'dogbyte',
  textAlign: 'center',
  color: "#ecf0f1",
  fontSize: 50
},
counterLabel: {
  fontFamily: 'dogbyte',
  textAlign: 'center',
  color: '#bbb',
  fontSize: 20,
  textTransform: "uppercase"
},
bestScoreContainer: {
  marginTop: 10,
  flexDirection: 'row',
  justifyContent: 'center'
},
bestScoreIcon: {
  width: 25,
  height: 25,
  marginRight: 5
},
bestScoreLabel: {
  fontFamily: 'dogbyte',
  color: '#bbb',
  fontSize: 25,
  marginTop: 2.5,
},
});