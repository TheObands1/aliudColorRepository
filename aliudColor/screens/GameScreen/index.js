import React, {useState, useEffect } from "react";
import { View, Text, Image, TouchableOpacity,  Dimensions  } from "react-native";
import { GameTitle, GameGrid} from "../../components";
import { BottomArea } from "../../components/GameplayComponents/bottomComponents/BottomArea";
import { PauseView } from "../../components/GameplayComponents/pauseComponents/pauseView";
import { LoseView } from "../../components/GameplayComponents/LoseView";
import GameStyle from "./styles";
import { HSLNormalColor, HSLDifferentColor } from '../../utilities'
import * as SQLite from 'expo-sqlite';

export default function GameView({route, navigation}) {

  const {isAccesibilityModeOn} = route.params;

  const [points, setPoints] = useState(0);
  const [currentMaxPoints, setCurrentMaxPoints] = useState(0);
  const [gridSize, setGridSize] = useState(2);
  const [timeLeft, setTimeLeft] = useState(10);
  const [timeToSubstract, setTimeToSubstract] = useState(3);
  const [currentColor, setCurrentColor] = useState(HSLNormalColor());
  const [currentGridMargin, setCurrentGridMargin] = useState(2);
  const [differentTilePosition, setDifferentTilePosition] = useState([]);
  const [differentTileColor, setDifferentTileColor] = useState();
  const [gameState, setGameState] = useState("Playing")
  const windowWidth = Dimensions.get('window').width;
  const gameDatabase = SQLite.openDatabase("gameDatabase");
  
  useEffect(() => {
    getCurrentMaxPoints();
    SetDefaultValues();
    startNewRound();
 }, [])

  const SavePoints  = () => {
    gameDatabase.transaction(tx => {
      tx.executeSql('INSERT INTO Scores (Score) values (?)', [points],
        (txObj, resultSet) => console.log("Insert was successful"),
        (txObj, error) => console.log('Error in insert', error))
    })
  }
    
 //Start Timer
  useEffect(() => {
    const interval = setInterval(() => {
      if(gameState === "Playing")
      {
        if(timeLeft-timeToSubstract <= 0)
        {
          setTimeLeft(0);
          setGameState("Lost");
          //Save points in DB
          SavePoints();
          if(points > currentMaxPoints)
          {
            setCurrentMaxPoints(points);
          }

        }
        else
        {
          setTimeLeft(prevTime => prevTime-timeToSubstract);
        }
      }
      
    }, 1000);
    return () => clearInterval(interval);
  }, [gameState, timeLeft]);

  function getCurrentMaxPoints(){
    gameDatabase.transaction(tx => {
      tx.executeSql('SELECT MAX(Score) FROM Scores', null, // passing sql query and parameters:null
        // success callback which sends two things Transaction object and ResultSet Object
        (txObj, { rows: { _array } }) => setCurrentMaxPoints(_array[0]["MAX(Score)"]), 
        // failure callback which sends two things Transaction object and Error
        (txObj, error) => console.log('This Error ', error)
        ) // end executeSQL
    }) // end transaction
}

  function SetDefaultValues(){
    if(isAccesibilityModeOn)
    {
      setTimeToSubstract(1);
      setTimeLeft(20);
    }
    else
    {
      setTimeToSubstract(3);
      setTimeLeft(10);
    }
    setGridSize(2);
    setCurrentGridMargin(2);
    setPoints(0);
    setGameState("Playing");
  }

  function GetRandomGridPosition(size) {
    return Math.floor(Math.random()*size);
  }

  function getNewGridSize(currentPoints){
    let newSize = Math.min(Math.max(Math.floor(Math.sqrt(currentPoints)), 2), 7);
    setGridSize(newSize)
  }

  function startNewRound() {
    setCurrentColor(HSLNormalColor());
    let newRGB = HSLDifferentColor(currentColor);
    setDifferentTileColor(`hsl(${newRGB.h}, ${newRGB.s}%, ${newRGB.l}%)`);
    getNewGridSize(points);
    setDifferentTilePosition([GetRandomGridPosition(gridSize),GetRandomGridPosition(gridSize)]);
  }

  const checkPressedTile = (row, column) => {
    if(row === differentTilePosition[0] && column == differentTilePosition[1])
    {
      setPoints(prevPoints => prevPoints+1);
      setTimeLeft(prevTime => prevTime+1);
      setCurrentGridMargin(prevMargin => prevMargin+0.01)
      startNewRound();
    }
    else
    {
      if(timeLeft-timeToSubstract < 0)
      {
        setTimeLeft(0);
      }
      else
      {
        setTimeLeft(prevTime => prevTime-2);
      }
    }
  }

  const changeGameState = () => {
    if(gameState === "Playing")
    {
      setGameState("Paused");
      return;
    }
    if(gameState === "Paused")
    {
      setGameState("Playing")
      return;
    }
    if(gameState === "Lost")
    {
      SetDefaultValues();
      startNewRound();
      setGameState("Playing")
      return;
    }
  }

  function currentGameView() {
    switch(gameState)
    {
      case "Playing":
        return(
                <GameGrid gridSize = {gridSize} 
                          differentTilePosition = {differentTilePosition} 
                          differentTileColor = {differentTileColor} 
                          currentColor = {currentColor} 
                          currentGridMargin = {currentGridMargin} 
                          checkPressedTile ={checkPressedTile} />
        );
      case "Paused":
        return(<PauseView/>);
      case "Lost":
        return (<LoseView/>);
      default:
        return;
    }
  }

  function goBackToMainMenu(){
    navigation.navigate("MainMenu");
  }


    return (
        <View style={GameStyle.backgroundStyle}>
          <View style ={{flex: 1, justifyContent: 'center'}}>
            <GameTitle fontSize={30}/>
          </View>
          <View style = {{ flex: 3, justifyContent: 'center' }}>
            <View style = {{ flex: 1, height: windowWidth / 2, width: windowWidth / 1.2, flexDirection: 'row'}}>
                {
                  currentGameView()
                }
            </View>
          </View>
          <View style={{ flex: 2 }}>
            <BottomArea gameState = {gameState} 
                        changeGameState = {changeGameState}  
                        points = {points} 
                        timeLeft = {timeLeft} 
                        windowWidth = {windowWidth}
                        goBackToMainMenu = {goBackToMainMenu}
                        currentMaxPoints = {currentMaxPoints}
                        />
          </View>
        </View>
      );

}

