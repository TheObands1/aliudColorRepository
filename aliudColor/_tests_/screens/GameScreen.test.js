import React from 'react';
import renderer from 'react-test-renderer';
import GameScreen from  '../../screens/GameScreen/index'
import { GameGrid } from '../../components'
describe('<GameScreen /> tests', () => {
  /*Although this test passes, because every round the grid color in the gamescreen is different, 
  running this test normally won't do it. You need to update it every time*/
    it('Render the component correctly ', () => {
    
        const route = { params: { isAccesibilityModeOn: false } }
        const navigate = jest.fn();
        const tree = renderer.create(<GameScreen route= {route} navigation={{ navigate }} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    
});
