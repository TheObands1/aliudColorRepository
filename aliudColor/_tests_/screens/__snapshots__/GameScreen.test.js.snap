// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameScreen /> tests Render the component correctly  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0a0a0a",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityLabel="aliudColor"
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#ecf0f1",
              "fontSize": 30,
            },
          ]
        }
      >
        aliud
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#E64C3C",
              "fontSize": 30,
              "textTransform": "uppercase",
            },
          ]
        }
      >
        c
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#E57E31",
              "fontSize": 30,
            },
          ]
        }
      >
        o
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#F1C431",
              "fontSize": 30,
            },
          ]
        }
      >
        l
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#68CC73",
              "fontSize": 30,
            },
          ]
        }
      >
        o
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#3998DB",
              "fontSize": 30,
            },
          ]
        }
      >
        r
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "height": 375,
          "width": 625,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
        testID="gridColumn"
      >
        <View
          accessibilityLabel=" Wrong!"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "hsl(113, 91%, 67%)",
              "flex": 1,
              "margin": 2,
              "opacity": 1,
            }
          }
          testID="gridTile"
        />
        <View
          accessibilityLabel=" Wrong!"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "hsl(113, 91%, 67%)",
              "flex": 1,
              "margin": 2,
              "opacity": 1,
            }
          }
          testID="gridTile"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
        testID="gridColumn"
      >
        <View
          accessibilityLabel=" Wrong!"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "hsl(113, 91%, 67%)",
              "flex": 1,
              "margin": 2,
              "opacity": 1,
            }
          }
          testID="gridTile"
        />
        <View
          accessibilityLabel=" Wrong!"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "hsl(113, 91%, 67%)",
              "flex": 1,
              "margin": 2,
              "opacity": 1,
            }
          }
          testID="gridTile"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "width": 656.25,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": "auto",
            "marginTop": "auto",
          }
        }
      >
        <View
          accessible={true}
        >
          <Text
            style={
              Object {
                "color": "#ecf0f1",
                "fontFamily": "dogbyte",
                "fontSize": 50,
                "textAlign": "center",
              }
            }
          >
            0
          </Text>
          <Text
            style={
              Object {
                "color": "#bbb",
                "fontFamily": "dogbyte",
                "fontSize": 20,
                "textAlign": "center",
                "textTransform": "uppercase",
              }
            }
          >
            points
          </Text>
        </View>
        <View
          accessibilityLabel="Current High Score0"
          accessible={true}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 25,
                "marginRight": 5,
                "width": 25,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#bbb",
                "fontFamily": "dogbyte",
                "fontSize": 25,
                "marginTop": 2.5,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": "auto",
            "marginTop": "auto",
          }
        }
      >
        <View
          accessibilityLabel="Pause Game Button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": "auto",
            "marginTop": "auto",
          }
        }
      >
        <View
          accessible={true}
        >
          <Text
            style={
              Object {
                "color": "#ecf0f1",
                "fontFamily": "dogbyte",
                "fontSize": 50,
                "textAlign": "center",
              }
            }
          >
            10
          </Text>
          <Text
            style={
              Object {
                "color": "#bbb",
                "fontFamily": "dogbyte",
                "fontSize": 20,
                "textAlign": "center",
                "textTransform": "uppercase",
              }
            }
          >
            time left
          </Text>
        </View>
        <View
          accessibilityLabel="Time Icon"
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 25,
                "marginRight": 5,
                "width": 25,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
